waf.go

```diff
@@ -35,14 +35,51 @@ type Diagnostics struct {
 	Scanners       *DiagnosticEntry
 }
 
+// TopLevelErrors returns the list of top-level errors reported by the WAF on any of the Diagnostics
+// entries, rolled up into a single error value. Returns nil if no top-level errors were reported.
+// Individual, item-level errors might still exist.
+func (d *Diagnostics) TopLevelError() error {
+	fields := map[string]*DiagnosticEntry{
+		"rules":          d.Rules,
+		"custom_rules":   d.CustomRules,
+		"exclusions":     d.Exclusions,
+		"rules_override": d.RulesOverrides,
+		"rules_data":     d.RulesData,
+		"processors":     d.Processors,
+		"scanners":       d.Scanners,
+	}
+
+	errs := make([]error, 0, len(fields))
+	for field, entry := range fields {
+		if entry == nil || entry.Error == "" {
+			// No entry or no error => we're all good.
+			continue
+		}
+		errs = append(errs, fmt.Errorf("in %#v: %s", field, entry.Error))
```

Comment:
It's possible to leverage an API like https://pkg.go.dev/errors#Join too for instance (our Go min version prevents us from using it)

Reply:
I know... Made this because I think it's a better DevX in case of issue at this stage...